feat: add slowDown decorator to fastify (#13)
nearform/fastify-slow-down: README.md

## Usage

Register SlowDown as a Fastify plugin.
This plugin will add an `onRequest` hook to slow down replies if a client (based on their IP address by default) has made too many multiple requests in the given `timeWindow` and

> it will add `slowDown` decoration to the `request` with the following information:

Reviewer: ```suggestion
it will add `slowDown` request decorator, which is an object with the following properties:
```

| Name | Type | Description |
|------|------|-------------|
| limit | number | the maximum limit until the server starts to delay the response |
| index request | number | the index of the current request|
| remaining requests | number | how many requests are left until the server starts to delay the response |

> | delay | number, undefined | value of the delay (in milliseconds) applied to this request |

Reviewer: 1. use the property names, not bogus names!
2. use typescript syntax to describe the types, not an invented syntax
